src/discovery/erc_4626.rs

```diff
@@ -12,14 +12,16 @@ use crate::{
     errors::AMMError,
 };
 
+use super::discovery_options::Erc4626DiscoveryOptions;
+
 lazy_static::lazy_static! {
     static ref HEX_REGEX: Regex = Regex::new(r"0x[0-9a-fA-F]+").expect("Could not compile regex");
 }
 
 // Returns a vec of empty factories that match one of the Factory interfaces specified by each DiscoverableFactory
 pub async fn discover_erc_4626_vaults<M: Middleware>(
     middleware: Arc<M>,
-    step: u64,
+    options: Erc4626DiscoveryOptions,
```

Comment:
Can we make this optional, and use a default implementation if None is provided?

Reply:
fixed